docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,208 @@
+
+#### repair
+Repair loops over the metainfoDB and checks if a segment is injured, if it is injured, its added to the repair queue and a repairer later fixes it. If its unable to be repaired then its added to irreparabledb. The irreparable loop later iterates over this table and attempts to re-repair these injured segments. The repair system is currently made of 4 parts and 2 DBs (db tables). The 4 parts are 1) repair observer (contains ReliabilityCache) 2) irreparable loop 3) repairer 4) repair queue. The 2 DBs are 1) injuredsegment (repair queue) table in satellite.DB 2) and irreparabledb table in satellite.DB 
+
+#### garbage collection (GC)
+GC consults with the metainfoDB, the source of truth for what data should be on the network, then with this info creates lists of what data each storage node should have and sends those lists to their respective storgae node.  The storage nodes can then delete any data that they are storing that is no longer in the GC list. See [GC design doc](https://github.com/storj/storj/blob/master/docs/design/garbage-collection.md) for more details. 
+
+#### accounting
+Accounting includes tally and rollup service along with the live accounting cache. The function of accounting is to calculate what to charge Uplinks for their usage and also what to pay storage nodes for their contributions to the network. See [docs on accounting](https://github.com/storj/docs/blob/master/code/payments/Accounting.md) for more details.
+
+#### console
+Console provides the web UI for the Satellite where users can create new accounts/projects/apiKeys needed for uploading/downloading to the network.
+
+#### mail
+Mail sends email for the console UI.
+
+#### marketing
+Marketing provides a private web UI for the the marketing admins for referral program.
+
+#### nodestats
+Nodestats makes it so that storagenodes can ask the satellite for info about itself, for example, it can ask for stats on reputation and accounting storage usage.
+
+#### inspectors
+Inspectors are exposed on a private port and provide a way to get data about certain systems. The following inspectors currently exist: overlay inspector, health inspector, and irreparable inspector.
+
+#### kademlia
+Kademlia, discovery, bootstrap, and vouchers are being removed and not included in this doc. See [kademlia removal design doc](https://github.com/storj/storj/blob/master/docs/design/kademlia-removal.md) for more details.
+
+#### PRC endpoints
+The Satellite has the following RPC endpoints:
+- Public: metainfo, nodestats, orders, overlay (currently part of kademlia, but may be added here)
+- Private: inspectors
+
+#### HTTP endpoints
+The Satellite has the following HTTP endpoints:
+- Public: console
+- Private: marketing admin UI, debug
+
+#### databases
+All services (except version) make connections to the satellite.DB. Five services rely on metainfo service to access the metainfoDB, this includes inspectors, accounting, audit, repair, and garbage collection.
```

Comment:
All services use `satellite.DB`. Do all of them need the full set of databases?

Reply:
satellite.DB is the master interface, which contains all the "sub-databases", so in that sense yeah, all of them require some part of it... I was wondering whether it makes sense to clarify which parts what needs... but now to think about it... probably not too relevant.